backend/app/api/api_v1/endpoints/domains.py

```
@@ -3118,6 +3177,73 @@ async def get_domain_stats(
+    domain = workspace_domain_query(db, workspace).filter(Domain.name == domain_name).first()
+    if domain is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Domain not found",
+        )
+    token = _ensure_domain_verification_token(db, domain)
+    expected = _ownership_record_value(token)
+    record_name = _ownership_record_name(domain.name)
+    try:
+        observed = await get_default_provider(db).lookup_txt(record_name)
+    except Exception as exc:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)